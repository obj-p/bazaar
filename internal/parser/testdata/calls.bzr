package calls

func Calls() {
        Print("Hello, world!")
        BlockLiteral({})
        NamedArgs(first = 1337, second = "second")
        TrailingBlockLiteral {}
}
