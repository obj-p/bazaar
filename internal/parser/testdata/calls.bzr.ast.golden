&parser.Bazaar{
  Package: &parser.PackageDecl{
    Domain: []string{
      "calls",
    },
  },
  Decls: []*parser.TopLevelDecl{
    {
      Function: &parser.FunctionDecl{
        Name: "Calls",
        Block: []*parser.Stmt{
          {
            Callable: &parser.CallableStmt{
              Name: &"Print",
              Arguments: []*parser.ArgumentExpr{
                {
                  Expr: &parser.Expr{
                    Unary: &parser.UnaryExpr{
                      Primary: &parser.PrimaryExpr{
                        Literal: &parser.Literal{
                          String: &parser.String{
                            Fragments: []*parser.StringFragment{
                              {
                                Text: &"Hello, world!",
                              },
                            },
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          },
          {
            Callable: &parser.CallableStmt{
              Name: &"BlockLiteral",
              Arguments: []*parser.ArgumentExpr{
                {
                  Expr: &parser.Expr{
                    Unary: &parser.UnaryExpr{
                      Primary: &parser.PrimaryExpr{
                        Lambda: &parser.LambdaExpr{
                        },
                      },
                    },
                  },
                },
              },
            },
          },
          {
            Callable: &parser.CallableStmt{
              Name: &"NamedArgs",
              Arguments: []*parser.ArgumentExpr{
                {
                  Name: &"first",
                  Expr: &parser.Expr{
                    Unary: &parser.UnaryExpr{
                      Primary: &parser.PrimaryExpr{
                        Literal: &parser.Literal{
                          Number: &"1337",
                        },
                      },
                    },
                  },
                },
                {
                  Name: &"second",
                  Expr: &parser.Expr{
                    Unary: &parser.UnaryExpr{
                      Primary: &parser.PrimaryExpr{
                        Literal: &parser.Literal{
                          String: &parser.String{
                            Fragments: []*parser.StringFragment{
                              {
                                Text: &"second",
                              },
                            },
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          },
          {
            Callable: &parser.CallableStmt{
              Name: &"TrailingBlockLiteral",
              LambdaOnly: &parser.LambdaExpr{
              },
            },
          },
          {
            Callable: &parser.CallableStmt{
              Name: &"call",
              Arguments: []*parser.ArgumentExpr{
                {
                  Expr: &parser.Expr{
                    Unary: &parser.UnaryExpr{
                      Primary: &parser.PrimaryExpr{
                        Reference: &parser.ReferenceExpr{
                          Name: &"first",
                        },
                      },
                    },
                  },
                },
                {
                  Expr: &parser.Expr{
                    Unary: &parser.UnaryExpr{
                      Primary: &parser.PrimaryExpr{
                        Reference: &parser.ReferenceExpr{
                          Name: &"second",
                        },
                      },
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
  },
}
